biocrnpyler/chemical_reaction_network.py

```diff
@@ -232,6 +236,18 @@ def replace_species(self, species: Species, new_species: Species):
         
         return ComplexSpecies(species = new_species_list, name = new_name, material_type = self.material_type, attributes = self.attributes)
 
+    #Returns all species in the ComplexSpecies. If recursive = True, returns species inside internal ComplexSpecies recursively as well.
+    def get_species(self, recursive = False):
```

Comment:
Is there a strong case for recursive=True? I plan to change this to a more pythonic code in #103, namely:
```python
class ComplexSpecies:
     @property 
     def species(self):
         return self.species
```    
Then, you can use it like:
```python
cp = ComplexSpecies([sp1,sp2])
print(cp.species)
```
You don't need to change the code, that's not my point, just I want to know why recursive=True is needed right now.

Reply:
So I think this is a little different than just accessing ComplexSpecies.species.

In the non-recrusive case, only the ComplexSpecies is returned. In the recursive case, all the Species in the ComplexSpecies are returned and furthermore, if the Species inside the ComplexSpecies is also a ComplexSpecies, the species inside that ComplexSpecies are returned and so on recursively.

I think this actually makes 3 cases, which maybe we can have different functions for (or the same function with different keywords):

1. Get the list of species inside a ComplexSpecies, with no kind of recursion.
2. Get the list of species inside a ComplexSpecies with recursion (ie flatten all nested ComplexSpecies).
3. A base-case for 2 so when a Species is hit the recursive function stops.

In terms of this code, it is important to have both recursive and non-recursive calls in order to filter ComplexSpecies correctly in different cases when using GlobalMechanisms. I can also imagine other cases when we might want to flatten all nested ComplexSpecies.